scipy/optimize/tests/test_optimize.py

```diff
@@ -157,7 +157,9 @@ def test_bfgs_nan(self):
         fprime = lambda x: np.ones_like(x)
         x0 = [np.nan]
         with np.errstate(over='ignore', invalid='ignore'):
-            x = optimize.fmin_bfgs(func, x0, fprime, disp=False)
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', LineSearchWarning)
+                x = optimize.fmin_bfgs(func, x0, fprime, disp=False)
```

Comment:
But if we filter warnings in `_line_search_wolfe12 ` a user won't see them when this wrapper is called. It looks like undesired behaviour, doesn't it?

I might be confused with `LineSearchWarning` vs `_LineSearchError`. The former is raised in `line_search_wolfe2` when `derphi_star` is `None`, the latter is raised in `_line_search_wolfe12 ` when `alpha_star` is `None`.

As far as I understand we still want to give a warning (but not an error), when `derphi_star` is `None`. Although currently I have no idea about algorithm details.